packages/swapper/src/thorchain-utils/getL1RateOrQuote.ts

```diff
@@ -0,0 +1,430 @@
+    // Fees consist of liquidity, outbound, and affiliate fees
+    // For the purpose of displaying protocol fees to the user, we don't need the latter
+    // The reason for that is the affiliate fee is shown as its own "ShapeShift fee" section
+    // Including the affiliate fee here would result in the protocol fee being wrong, as affiliate fees would be
+    // double accounted for both in protocol fees, and affiliate fee
+    const buyAssetTradeFeeBuyAssetCryptoThorPrecision = bnOrZero(quote.fees.total).minus(
+      quote.fees.affiliate,
+    )
+
+    const buyAssetTradeFeeBuyAssetCryptoBaseUnit = convertPrecision({
+      value: buyAssetTradeFeeBuyAssetCryptoThorPrecision,
+      inputExponent: nativePrecision,
+      outputExponent: buyAsset.precision,
+    })
+
+    const protocolFees: Record<AssetId, ProtocolFee> = {}
+
+    if (!buyAssetTradeFeeBuyAssetCryptoBaseUnit.isZero()) {
+      protocolFees[buyAsset.assetId] = {
+        amountCryptoBaseUnit: buyAssetTradeFeeBuyAssetCryptoBaseUnit.toString(),
+        requiresBalance: false,
+        asset: buyAsset,
+      }
+    }
+
+    return protocolFees
+  }
+
+  const getMemo = (route: ThorTradeRoute) => {
+    if (input.quoteOrRate === 'rate') return ''
+
+    if (!route.quote.memo) throw new Error('no memo provided')
+
+    // always use auto stream quote memo (0 limit = 5bps - 50bps, sometimes up to 100bps)
+    // see: https://discord.com/channels/838986635756044328/1166265575941619742/1166500062101250100
+    if (route.isStreaming) return route.quote.memo
+
+    const limitWithManualSlippage = getLimitWithManualSlippage({
+      expectedAmountOutThorBaseUnit: route.expectedAmountOutThorBaseUnit,
+      slippageBps: route.slippageBps,
+    })
+
+    return addLimitToMemo({
+      memo: route.quote.memo,
+      limit: limitWithManualSlippage,
+      affilate: getAffiliate(swapperName),
+    })
+  }
+
+  const makeThorTradeRateOrQuote = <U extends ThorTradeRateOrQuote>({
+    route,
+    memo,
+    allowanceContract,
+    feeData,
+    data,
+    router,
+    vault,
+  }: MakeThorTradeInput<U>): T => {
+    const buyAmountAfterFeesCryptoBaseUnit = convertPrecision({
+      value: route.expectedAmountOutThorBaseUnit,
+      inputExponent: nativePrecision,
+      outputExponent: buyAsset.precision,
+    }).toFixed()
+
+    const rate = getRouteRate(route.expectedAmountOutThorBaseUnit)
+    const slippage = route.isStreaming ? undefined : slippageTolerancePercentageDecimal
+    const buyAmountBeforeFeesCryptoBaseUnit = getRouteBuyAmountBeforeFeesCryptoBaseUnit(route.quote)
+
+    return {
+      id: uuid(),
+      quoteOrRate: input.quoteOrRate,
+      memo,
+      receiveAddress,
+      affiliateBps: route.affiliateBps,
+      isStreaming: route.isStreaming,
+      recommendedMinimumCryptoBaseUnit,
+      slippageTolerancePercentageDecimal: slippage,
+      rate,
+      data,
+      router,
+      vault,
+      expiry: route.quote.expiry,
+      tradeType: TradeType.L1ToL1,
```

Comment:
Similarly here, shouldn't this be `tradeType: tradeType ?? TradeType.L1ToL1`, where `tradeType` is an arg of `getL1RateOrQuote`? See https://github.com/shapeshift/web/pull/9623/files#r2128326565